sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -0,0 +1,154 @@
+import os
+import urllib
+
+from azure.core.pipeline.policies import HttpLoggingPolicy
+import logging
+import types
+
+
+
+class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
+
+    def __init__(self, logger=None, **kwargs):
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+        super().__init__(logger, **kwargs)
+        if self._enable_diagnostics_logging:
+            self.logger = logger or logging.getLogger(__name__)
+
+    def on_request(self, request):  # type: (PipelineRequest) -> None
+        http_request = request.http_request
+        options = request.context.options
+        self._enable_diagnostics_logging = request.context.setdefault(
+            "enable_diagnostics_logging",
+            options.pop("enable_diagnostics_logging", self._enable_diagnostics_logging))
+        if self._enable_diagnostics_logging:
+            # Get logger in my context first (request has been retried)
+            # then read from kwargs (pop if that's the case)
+            # then use my instance logger
+            logger = request.context.setdefault("logger", options.pop("logger", self.logger))
+
+            if not logger.isEnabledFor(logging.INFO):
+                return
+
+            try:
+                parsed_url = list(urllib.parse.urlparse(http_request.url))
```

Comment:
We shouldn't duplicate this section of code - could you point out where this deviates from the original implementation so we can figure out the best way to accommodate it?

Reply:
Same comment applies to response handling.